### ISSUE TITLE
Lint/NameTypo reports false-positives for Time.now with ProjectIndexIncludesGems: true

### ISSUE THREAD

**Issue by Aqualon** [rubocop/rubocop#15643]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/NameTypo

### Steps to reproduce the problem

Repo to reproduce: https://github.com/Aqualon/test_nametypo/tree/main

Enable rubydex for rubocop including gems
```
AllCops:
  UseProjectIndex: true
  ProjectIndexIncludesGems: true
```

With rails in Gemfile
```ruby
bundle exec rubocop --only 'Lint/NameTypo' time.rb
```
reports
```ruby
time.rb:1:11: W: Lint/NameTypo: Possible typo: Time does not respond to now. Did you mean noon?
puts Time.now.utc
```
once rails is installed in Gemfile. With rails commented out it works
```
❯ bundle exec rubocop --only 'Lint/NameTypo' time.rb 
Resolving dependencies...
Inspecting 1 file
.

1 file inspected, no offenses detected
```

### Expected behavior

The cop should not report Time.now as missing method.

### Actual behavior

Cop reports `Possible typo: Time does not respond to now. Did you mean noon?`

### RuboCop configuration

```yaml
AllCops:
  NewCops: enable
  UseProjectIndex: true
  ProjectIndexIncludesGems: true
```

### RuboCop version

```shell
❯ rubocop -V
1.90.0 (using Parser 3.3.12.0, rubocop-ast 1.50.0, analyzing as Ruby 2.7, running on ruby 4.0.6) +Rubydex [arm64-darwin23]
```

**Comment by Aqualon** [comment]
In a bigger private repo I also got complaints about
```
Lint/NameTypo: Possible typo: RSpec does not respond to shared_examples. Did you mean clear_examples?
```
even so that's the documented way of using shared_examples on https://rspec.info/features/3-13/rspec-core/example-groups/shared-examples/

and one for `Regexp.new` that also exists https://docs.ruby-lang.org/en/4.0/Regexp.html#method-c-new
```
Lint/NameTypo: Possible typo: Regexp does not respond to new. Did you mean next?
```
This one also is reported with just rubocop/rubydex and no other gems in Gemfile
